core/src/main/java/org/aerogear/mobile/core/metrics/publisher/NetworkMetricsPublisher.java

```diff
@@ -0,0 +1,55 @@
+package org.aerogear.mobile.core.metrics.publisher;
+
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.metrics.Metrics;
+import org.aerogear.mobile.core.metrics.MetricsPublisher;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+/**
+ * Sends metrics data to the backend using the configuration in JSON config file
+ */
+public class NetworkMetricsPublisher implements MetricsPublisher {
+
+    private HttpRequest httpRequest;
+    private String url;
+
+    public NetworkMetricsPublisher(HttpRequest httpRequest, String url) {
+        this.httpRequest = httpRequest;
+        this.url = url;
+    }
+
+    @Override
+    public void publish(Metrics... metrics) {
+
+        try {
+
+            JSONObject json = new JSONObject();
+            for (final Metrics m : metrics) {
+                json.put(m.identifier(), new JSONObject(m.data()));
+            }
+
+            httpRequest.post(url, json.toString().getBytes());
+
+            MobileCore.getLogger().debug("Sending metrics");
+            HttpResponse httpResponse = httpRequest.execute();
+            httpResponse.onComplete(() -> {
+                if (httpResponse.getStatus() == HTTP_OK) {
+                    MobileCore.getLogger().debug("Metrics sent");
+                } else {
+                    MobileCore.getLogger().error(httpResponse.getRequestError().getMessage(),
```

Comment:
There is really no way how consumer of the API can know if some error happened. Shouldn't be there some way, like optionally adding callback on error.
```
public void publish(Metrics... metrics, PushFailedCallback errorCallback) {
```

Reply:
@wojta at the moment the metrics are automatically sent by the core sdk, but when we add custom user metrics we should provide such a callback.

Reply:
I can live with that now :)